nixos/modules/services/infrastructure/terraform-cloud-agent/default.nix

```
@@ -0,0 +1,65 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.terraform-cloud-agent;
+in
+{
+  options.services.terraform-cloud-agent = {
+    enable = lib.mkEnableOption "terraform-cloud-agent";
+
+    name = lib.mkOption {
+      type = lib.types.str;
+    };
+
+    log-level = lib.mkOption {
+      type = lib.types.nullOr (lib.types.enum [ "trace" "debug" "info" "warn" "error" ]);
+      default = null;
+    };
+
+    data-dir = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+    };
+
+    cache-dir = lib.mkOption {
+      type = lib.types.nullOr lib.types.path;
+      default = null;
+    };
+
+    address = lib.mkOption {
+      type = lib.types.nullOr lib.types.str;
+      default = null;
+    };
+
+    tokenPath = lib.mkOption {
+      type = lib.types.path;
+      default = "/etc/terraform-cloud-agent-token";
+    };
+
+    accept = lib.mkOption {
+      type = lib.types.commas;
+      default = "plan,apply,policy,assessment";
+    };
+  };
+
+  config.systemd.services.terraform-cloud-agent = lib.mkIf cfg.enable {
+    description = "Agent that executes plans from Terraform Cloud";
+    wantedBy = [ "multi-user.target" ];
+    after = [ "network.target" ];
+
+    script = ''
+      export TFC_AGENT_TOKEN=$(cat ${cfg.tokenPath})
+      export TFC_AGENT_DATA_DIR=''${STATE_DIRECTORY:-${cfg.data-dir}}
+      export TFC_AGENT_CACHE_DIR=''${CACHE_DIRECTORY:-${cfg.cache-dir}}
```

Ditto the [specifiers](https://www.man7.org/linux/man-pages/man5/systemd.unit.5.html#SPECIFIERS) here - they can be referred to in environment variables and will be pasted in:
```suggestion
    environment = {
      TFC_AGENT_DATA_DIR = "%S";
      TFC_AGENT_CACHE_DIR = "%C";
    };
    
    script = ''
      export TFC_AGENT_TOKEN=$(cat ${cfg.tokenPath})
```

---

Addressed in 0ffe3d1a6d69.